### ISSUE TITLE
Problem sending over ipv6

### ISSUE THREAD

**Issue by adityasingh007** [zone-eu/zone-mta#79]
I want ZMTA's senders to connect over ipv6. Are my configs correct ? What is ignoreIPv6 used for ? The mails are not going over ipv6.

```
    "pools": {
        "default": [{
            "address": "MYIPV6ADDRESS",
            "name": "MYSENDINGDOMAIN"
        }]
    },


    "zones": {
        "default": {
            "processes": 1,
            "preferIPv6" : true,
            "ignoreIPv6" : false,
            "connections": 1,
            "throttling": "1 messages/minute",
            "pool": "default"
        },

```

```
verb HTTP /metrics
verb HTTP /metrics
info SMTP/feeder/2035 Connection from [IPOFMYMAINSERVER]:52906
info Queue/2035 15c2bec4016000a5c6 QUEUED (message-id=<f2a3fb479ba4ceef9055e4167fadf533@MYSENDINGDOMAIN> from=newsletter@MYSENDINGDOMAIN to=something@mail-tester.com source=IPOFMYMAINSERVER subject="Your Shipment Is On The Way" body=7616 md5=5035cd035db7)
verb Queue 15c2bec4016000a5c6.001 SHIFTED (key="lock 15c2bec4016000a5c6 001" zone="default")
verb Sender/default/2036 15c2bec4016000a5c6.001 FETCHED for delivery (from=newsletter@MYSENDINGDOMAIN to=something@mail-tester.com)
info Sender/default/2036 15c2bec4016000a5c6.001 CONNECTING domain=mail-tester.com port=25
info Sender/default/2036 15c2bec4016000a5c6.001 CONNECTED domain=mail-tester.com mx=mail-tester.com[94.23.206.89]
verb Sender/default/2036 15c2bec4016000a5c6.001 INITIALIZING [B29LQuE6y0] (mx=94.23.206.89 mta=0.0.0.0)
verb Sender/default/2036 15c2bec4016000a5c6.001 INITIALIZED [B29LQuE6y0] (mx=94.23.206.89 mta=0.0.0.0)
verb Sender/default/2036 15c2bec4016000a5c6.001 [B29LQuE6y0] CONNECTED mx=mail-tester.com[94.23.206.89]
verb Sender/default/2036 15c2bec4016000a5c6.001 SENDING [B29LQuE6y0] (from=newsletter@MYSENDINGDOMAIN to=something@mail-tester.com size=9143)
info Sender/default/2036 15c2bec4016000a5c6.001 ACCEPTED from=newsletter@MYSENDINGDOMAIN[newsletter@MYSENDINGDOMAIN] to=something@mail-tester.com mx=mail-tester.com body=7616 md5=5035cd035db7[MD5_OK] (250 2.0.0 Ok: queued as 5ED1E9F7B3)
info Sender/default/2036 15c2bec4016000a5c6.001 SMTPCLOSE [B29LQuE6y0] Connection closed response="250 2.0.0 Ok: queued as 5ED1E9F7B3"
verb Queue Deleting unreferenced message 15c2bec4016000a5c6
verb Queue 15c2bec4016000a5c6 REMOVE
verb HTTP /metrics
verb HTTP /metrics
verb HTTP /metrics
verb HTTP /metrics

```

**Comment by andris9** [comment]
If `preferIPv6` option is true, then ZoneMTA resolves AAAA record first for a MX domain and if it is not found, fallbacks to A. If `preferIPv6` is false, the ZoneMTA tries to first resolve A and then AAAA. If `ignoreIPv6` is true, then ZoneMTA does not resolve the AAAA record even if A is not found.

The server you are trying to send (mail-tester.com) does not have an AAAA record and thus IPv6 can not be used for that server.

**Comment by adityasingh007** [comment]
Alas. It seems to be the same for gmail.

`
verb Queue 15c2c65335f000f266.001 SHIFTED (key="lock 15c2c65335f000f266 001" zone="default")
verb Sender/default/2114 15c2c65335f000f266.001 FETCHED for delivery (from=newsletter@SENDINGDOMAIN to=aditya@impulsemedia.co.in)
info Sender/default/2114 15c2c65335f000f266.001 CONNECTING domain=impulsemedia.co.in port=25
info Sender/default/2114 15c2c65335f000f266.001 CONNECTED domain=impulsemedia.co.in mx=aspmx.l.google.com[74.125.29.27]
verb Sender/default/2114 15c2c65335f000f266.001 INITIALIZING [5DLjDyoFW3k] (mx=74.125.29.27 mta=0.0.0.0)
verb Sender/default/2114 15c2c65335f000f266.001 INITIALIZED [5DLjDyoFW3k] (mx=74.125.29.27 mta=0.0.0.0)
verb Sender/default/2114 15c2c65335f000f266.001 [5DLjDyoFW3k] CONNECTED mx=aspmx.l.google.com[74.125.29.27]
verb Sender/default/2114 15c2c65335f000f266.001 SENDING [5DLjDyoFW3k] (from=newsletter@SENDINGDOMAIN to=aditya@impulsemedia.co.in size=7869)
verb HTTP /metrics
info Sender/default/2114 15c2c65335f000f266.001 ACCEPTED from=newsletter@SENDINGDOMAIN[newsletter@SENDINGDOMAIN] to=aditya@impulsemedia.co.in mx=aspmx.l.google.com body=6546 md5=3bcfc981d967[MD5_OK] (250 2.0.0 OK 1495393454 c2si15740258qtd.145 - gsmtp)
info Sender/default/2114 15c2c65335f000f266.001 SMTPCLOSE [5DLjDyoFW3k] Connection closed response="250 2.0.0 OK 1495393454 c2si15740258qtd.145 - gsmtp"
v
`

**Comment by andris9** [comment]
Could you try to upgrade ZoneMTA to latest commit and the try again?
